automatic/nexus-repository/tools/helpers.ps1

```diff
@@ -39,58 +38,122 @@ function Restore-NexusSSL {
+    $Config = Get-NexusConfiguration -Path $ConfigPath -ErrorAction SilentlyContinue
+
+    if ($Config.'application-port-ssl' -gt 0) {
+      # This is to combat Package Internalizer's over-enthusiastic URL matching
+      ('http' + 's')
+      if ($CertDomain = Get-NexusCertificateDomain -ConfigPath $ConfigPath) {
+      if (-not $script:OverriddenDomains) {$script:OverriddenDomains = @{}}
+        if ($CertDomain -notmatch '^\*') {
+          $CertDomain
+        } elseif ($CertDomain -match '^\*' -and $HostnameOverride -like $CertDomain) {
+          ($script:OverriddenDomains[$CertDomain] = $HostnameOverride)
+        } elseif ($CertDomain -match '^\*') {
+          while ($script:OverriddenDomains[$CertDomain] -notlike $CertDomain) {
+            $script:OverriddenDomains[$CertDomain] = Read-Host "Please provide the FQDN for Nexus matching the '$($CertDomain)' certificate"
+          }
+          $script:OverriddenDomains[$CertDomain]
+        }
+      } else {
+        Write-Warning "Could not figure out SSL configuration for $($env:ComputerName)"
+        $env:ComputerName
+      }
+      $Config.'application-port-ssl'
+    } elseif ($Config.'application-port' -gt 0) {
+      'http'
+      if ($HostnameOverride) {
+        $HostnameOverride
+      } else {
+        "localhost"
+      }
+      $Config.'application-port'
+    }
+
+    $Config.'nexus-context-path'
   }
 
+  # Set defaults if still not present
+  if (-not $Hostname) {$Hostname = "localhost"}
+  if (-not $Scheme) {$Scheme = 'http'}
+  if (-not $Port) {$Port = '8081'}
+
+  "$($Scheme)://$($Hostname):$($Port)$($Path)"
+}
+
+function Wait-NexusAvailability {
+  param(
+    # The hostname in use
+    [Parameter()]
+    [string]$Hostname,
+
+    # The path to the config file
+    [Parameter(Mandatory)]
+    [Alias("Config")]
+    [string]$NexusConfigFile = (Join-Path $env:ProgramData "\sonatype-work\nexus3\etc\nexus.properties"),
+
+    # Configurable timeout in minutes
+    [uint16]$Timeout = 3
+  )
   # As the service is started, this should be present momentarily
   $Timer = [System.Diagnostics.Stopwatch]::StartNew()
   while (-not ($ConfigPresent = Test-Path $NexusConfigFile) -and $Timer.Elapsed.TotalSeconds -le 60) {
     Write-Verbose "Waiting for '$($NexusConfigFile)' to become available ($($Timer.Elapsed.TotalSeconds) seconds waited)..."
     Start-Sleep -Seconds 5
   }
 
-  if ($ConfigPresent) {
-    $nexusPort = (Get-Content $NexusConfigFile | Where-Object {
-        $_ -match $portConfigLine
-      }).Split('=')[-1]
-
-    $nexusPath = (Get-Content $NexusConfigFile | Where-Object {
-        $_ -match "nexus-context-path"
-      }).Split("=")[-1]
-  } else {
-    Write-Warning "Expected Nexus Config file '$($NexusConfigFile)' is not present."
-    $nexusPath, $nexusPort = '/', $Port
-  }
-
-  $NexusUri = "$($nexusScheme)://$($hostname):$($nexusPort)$($nexusPath)"
+  $NexusUri = Get-NexusUri -Hostname $Hostname
```

Comment:
```suggestion
  $NexusUri = Get-NexusUri -HostnameOverride $Hostname
```

Just wondering if this should be changed? Not sure if PowerShell is doing magic to make `-Hostname` match `HostnameOverride`